helloworld-impl/src/main/java/sample/helloworld/impl/TestImpl.java

```diff
@@ -14,26 +14,18 @@
 
 public class TestImpl implements Test {
 
-  //#inject-service
-  private final HelloService helloService;
-
-  @Inject public TestImpl(HelloService helloService) {
-    this.helloService = helloService;
-  }
-  //#inject-service
-
-  public ServiceCall<NotUsed, NotUsed> audit() {
-    //#subscribe-to-topic
-    helloService.greetingsTopic().subscribe() // <-- you get back a Subscriber instance
-        .atLeastOnce(
-            Flow.fromFunction(this::doSomethingWithTheMessage));
-    //#subscribe-to-topic
-    return name -> completedFuture(NotUsed.getInstance());
-  }
-
-  private Done doSomethingWithTheMessage(GreetingMessage message) {
-    System.out.println("Message: " + message);
-    return Done.getInstance();
-  }
+    private final HelloService helloService;
+
+    @Inject
+    public TestImpl(HelloService helloService) {
+        this.helloService = helloService;
+        helloService.greetingsTopic().subscribe()
+                .atLeastOnce(Flow.fromFunction(this::doSomethingWithTheMessage));
+    }
+
+    private Done doSomethingWithTheMessage(GreetingMessage message) {
+        System.out.println("Message: " + message);
+        return Done.getInstance();
+    }
 
```

Comment:
Code formatting made this change look bigger than it really is (converted 2-space into 4-space, sorry).

This TestServiceImpl and TestService are not even necessary. You could have a separate class that's instantiated from the `Module` whose sole responsibility is to start the consumer. See https://github.com/lagom/online-auction-java/blob/master/search-impl/src/main/java/com/example/auction/search/Module.java#L27